### ISSUE TITLE
there is a bug when using specials

### ISSUE THREAD

**Issue by sys1874** [pytorch/text#265]
![image](https://user-images.githubusercontent.com/19858774/38081371-7cfacd0e-3376-11e8-8fc4-00537c5c77aa.png)
 
could anyone solve this problem. 
I just want to let '<pad>' to 0

**Comment by mttk** [comment]
The `specials` argument of the `Vocab` class is set by itself -- you do not need to explicitly pass it (https://github.com/pytorch/text/blob/master/torchtext/data/field.py#L253).

By default, the padding token will be set to `<pad>` (check the arguments of the `data.Field` constructor), so everything will be OK if you just remove the specials argument from the `build_vocab` call.

**Comment by sys1874** [comment]
Thanks your answer,However, how to set pad to 0 by using build_vocab(),because,unk will be 0 .
there will be a way to solve the problem
（https://github.com/pytorch/text/pull/216/commits/2cbafa2e89e340266d8044e28b06bce327966d86）
but , i don't want to change the code.

**Comment by mttk** [comment]
Right, I see what you wanted. I don't see a clean way to do this as the `_default_unk_index` is hardcoded, as well as the location of the `<unk>` token.

A monkeypatch to this would be to change the `stoi` of the Vocab object after calling `build_vocab`. You can essentially do something like:

```python
old_stoi = TEXT.vocab.stoi
# create a new defaultdict which defaults to some other index, or assign zero to <PAD>
# by doing something like TEXT.vocab.stoi['<pad>'] = 0
TEXT.vocab.stoi = new_stoi
```
I'm _pretty_ sure this won't break anything, but it might, and it's ugly.

**Comment by sys1874** [comment]
@mttk  Thanks for your solution.I will have a try on it.

**Comment by tang1943** [comment]
_default_unk_index is hard code to zero, so  <unk> and <pad> are in conflict when changing TEXT.vocab.stoi
